master/pkg/stream/stream.go

```diff
@@ -26,15 +59,20 @@ type Event[T Msg] struct {
 type Streamer struct {
 	Cond *sync.Cond
 	// Msgs are pre-marshalled messages to send to the streaming client.
-	Msgs []*websocket.PreparedMessage
+	Msgs []interface{}
 	// Closed is set externally, and noticed eventually.
 	Closed bool
+	// PrepareFn is a user defined function that prepares Msgs for broadcast
+	PrepareFn func(interface{}) interface{}
 }
 
-func NewStreamer() *Streamer {
+func NewStreamer(prepareFn func(interface{}) interface{}) *Streamer {
```

Comment:
Have you considered the pros/cons of configuring `PrepareFn` on the Publisher instead of the Streamer?

Pros I can think of:
- Putting it on the publisher clearly indicates that it isn't allowed to be different from one Streamer to the next.

Cons I can think of:
- If you configure the publisher with the right PrepareFn, you probably still need to call the CollectStartupMessages routine with a matching PrepareFn or something.  So it might be nicer to have all of that handled in the `entrypoint` function.

Hm, well I was originally going to suggest that I wanted it on the publisher but actually I think I like it where it is better.

Reply:
Yeah I thought about these when I was writing the code. Here are some additional things I was thinking about for cons, mostly implementation related:
- if we choose to send data differently, setting it in the publisher would entail more logic in `core.go` to figure out which prepare function to use as an argument vs being defined in `system.go` or in the entrypoint
- The prepare function would have to be passed as an arg through more functions
- I thought the amount of work required for the publisher refactor would be substantially more (but thinking about that more, I think they're the same)

Also, I was wondering - even though technically shouldn't ever happen the way it is written, why shouldn't the preparation be allowed to be different from streamer to streamer?

Reply:
> even though technically shouldn't ever happen the way it is written, why shouldn't the preparation be allowed to be different from streamer to streamer?

Because it makes it less obvious what the intention of the code is.  If there's no reason in production why that would ever be the case, having it be an option is just confusing.